eclair-core/src/main/resources/reference.conf

```diff
@@ -99,11 +92,23 @@ eclair {
 
   revocation-timeout = 20 seconds // after sending a commit_sig, we will wait for at most that duration before disconnecting
 
-  auth-timeout = 10 seconds // will disconnect if connection authentication doesn't happen within that timeframe
-  init-timeout = 10 seconds // will disconnect if initialization doesn't happen within that timeframe
-  ping-interval = 30 seconds
-  ping-timeout = 10 seconds // will disconnect if peer takes longer than that to respond
-  ping-disconnect = true // disconnect if no answer to our pings
+  peer-connection {
+    auth-timeout = 10 seconds // will disconnect if connection authentication doesn't happen within that timeframe
+    init-timeout = 10 seconds // will disconnect if initialization doesn't happen within that timeframe
+    ping-interval = 30 seconds
+    ping-timeout = 10 seconds // will disconnect if peer takes longer than that to respond
+    ping-disconnect = true // disconnect if no answer to our pings
+
+    features = "0a8a" // initial_routing_sync + option_data_loss_protect + option_channel_range_queries + option_channel_range_queries_ex + variable_length_onion
```

Comment:
We would still need to pass them to the `PeerConnection` somehow. Maybe in the constructor? This would put the resolution (`features`/`overrideFeatures`) outside of the scope of `PeerConnection`, which tbh may be the proper way to do it. But by doing so, we move further from our standard pattern of defining actors `class ActorName(nodeParams: nodeParams, actor1: ActorRef, actor2: ActorRef...)`.

Reply:
That's where the `trait` approach could come in handy too.

Reply:
Yes I was thinking to resolve `features/overrideFeatures` outside `PeerConnection` after all it's a node-wide setting that affects many parts other than the `Peer` but i see the point that would break our class defining pattern. What if we leave `features` inside the `PeerConnectionConf` object but outside the `peer-connection` config block? It's a bit of an exception but lets us avoid doing more complex things with the trait approach and we don't break of pattern.

Reply:
I agree that `features` is a node-wide parameter, so imo it shouldn't be inside the peer conf. I think the trait approach would be nice, the only painful part is figuring out how to mix-in some common fields that would be at the top-level of the conf is "sub-traits" (like providing features along with a `PeerParams`).